### ISSUE TITLE
Akka HTTP can not read BodyPart when upload file name in Chinese

### ISSUE THREAD

**Issue by aiyanbo** [akka/akka#17032]
I have a route with handle file upload. like this:

```scala
  val uploadRoute = {
    path("upload") {
      post {
        entity(as[FormData]) { formData =>
          formData.parts.runForeach({ bodyPart =>
            // read a upload file, but not execute this block
            bodyPart.entity.dataBytes.runForeach({ byteString =>
              // read stream as a byte array
              byteString.toByteBuffer.array()
            })
          })
          complete("upload success!")
        }
      }
    }
  }
```

The following code sample would be read request entity:

```scala
  val uploadRoute = {
    path("upload") {
      post {
        extractRequest { request =>
          request.entity.dataBytes.runForeach({ byteString =>
            // read entity as a utf8 string
            byteString.utf8String
          })
          complete("upload success!")
        }
      }
    }
  }
```

**Comment by sirthias** [comment]
What error are you getting? (i.e.: What exactly do you mean by "can not read BodyPart"?)

**Comment by aiyanbo** [comment]
@sirthias  Just can not run `formData.parts.runForeach` function. Can not catch any errors.

**Comment by sirthias** [comment]
What do you mean by "can not run formData.parts.runForeach function"?
Does it compile? And if it does, are you getting an exception at run-time? If so, what is this exception? Can you show a stack trace?

**Comment by rkuhn** [comment]
Each of the `runForeach` returns a `Future[Unit]`, could you add an `onFailure` handler that logs possible exceptions?

**Comment by ktoso** [comment]
Hi @aiyanbo - could you provide us with an error (like @rkuhn explains in the above comment) so we can debug this problem?

It may well be solved for the upcoming 1.1 release already.
I'll close the ticket and will re-open once we get an exception we can investigate upon.

**Comment by ktoso** [comment]
We have another related ticket: https://github.com/akka/akka/issues/17847
And we believe it may have been fixed by https://github.com/akka/akka/pull/18580

Unless we get more input here we considering it fixed.
